project/AkkaDisciplinePlugin.scala

```diff
@@ -0,0 +1,191 @@
+    * methods not found in jdk8. To test whether this has the desired effect, compile akka-remote and check the
+    * invocation of 'ByteBuffer.clear()' in EnvelopeBuffer.class with 'javap -c': it should refer to
+    * "java/nio/ByteBuffer.clear:()Ljava/nio/Buffer" and not "java/nio/ByteBuffer.clear:()Ljava/nio/ByteBuffer":
+    * `scalacOptions in Compile ++= javaTarget`
+    */
+  def javaTarget(scalaBinary: String, fullJavaHomes: String): Seq[String] =
+    if (Versions.isJdk1dot)
+      Seq("-target:jvm-1.8")
+    else if (scalaBinary == "2.11")
+      Seq("-target:jvm-1.8", "-javabootclasspath", fullJavaHomes)
+    else
+      Seq("-release", "8", "-javabootclasspath", fullJavaHomes) // -release 8 is not enough, for some reason we need the 8 rt.jar explicitly #25330
+
+  def javaSource(fullJavaHomes: String): Seq[String] =
+    if (Versions.isJdk1dot)
+      Nil
+    else
+      Seq("-source", "8", "-target", "8", "-bootclasspath", fullJavaHomes)
+
+}
+
+object AkkaDisciplinePlugin extends AutoPlugin {
+
+  import scoverage.ScoverageKeys.{coverageFailOnMinimum, coverageHighlighting, coverageMinimum}
+  import wartremover._
+  import akka.Versions.isScalaMinor
+
+  override def trigger: PluginTrigger = allRequirements // noTrigger
+  override lazy val projectSettings = disciplineSettings
+
+  lazy val scoverageSettings = Seq(
+    coverageMinimum := 60, // TODO set during review
+    coverageFailOnMinimum := false,
+    coverageHighlighting := true)
+
+  /** As of v2.4.1 there are 37 built-in Warts.
+    * Module `akka-actor` currently requires 34 Warts removed.
+    * With Warts.all or Warts.unsafe, `akka-actor` has 3793 errors.
+    *
+    * Take on one at a time where possible.
+    * For some, with akka-actor vs typed, e.g. Wart.Any, just accept it and move on.
+    * To catch as many issues as we can in compile-time, we can also enable / disable
+    * by module as well, as a work-around.
+    */
+  lazy val wartremoverSettings = Seq(
+    wartremoverErrors in (Compile, compile) ++= Warts.allBut(
+      Wart.MutableDataStructures,
+      Wart.ExplicitImplicitTypes,
+      Wart.LeakingSealed,
+      Wart.Any,
+      Wart.AnyVal,
+      Wart.Var,
+      Wart.Throw,
+      Wart.Null,
+      Wart.FinalVal,
+      Wart.FinalCaseClass,
+      Wart.AsInstanceOf,
+      Wart.IsInstanceOf,
+      Wart.DefaultArguments,
+      Wart.Overloading,
+      Wart.Recursion,
+      Wart.NonUnitStatements,
+      Wart.ImplicitConversion,
+      Wart.ImplicitParameter,
+      Wart.Nothing,
+      Wart.PublicInference,
+      Wart.Equals,
+      Wart.Return,
+      Wart.Enumeration,
+      Wart.TraversableOps,
+      Wart.TryPartial,
+      Wart.OptionPartial,
+      Wart.Option2Iterable,
+      Wart.ArrayEquals,
+      Wart.ToString,
+      Wart.StringPlusAny,
+      Wart.While,
+      Wart.Product,
+      Wart.JavaSerializable,
+      Wart.Serializable),
+    wartremoverErrors in Test ++= Nil) // in progress
```

Comment:
Not refined yet and so far only compiled against Akka-actor. Will finish tomorrow.